src/main/java/frc/robot/Robot.java

```diff
@@ -337,47 +259,136 @@ public void testPeriodic() {
     teleopPeriodic();
   }
 
+  // retrieve heading from encoder; convert it from radians to degrees
+  public double getHeadingDeg() {
+    return (m_yawEncoder.getDistance()* 180.0 / Math.PI) / 10.0;
+  }
+
+  // get wheel translation in RPM
+  public double motorRPMsToWheelRPM(double aMotorRPM, double bMotorRPM) {
+    // TODO: describe the math that got us here
+    double wheelRPM = ((aMotorRPM - bMotorRPM) / (GEAR_RATIO_123 * 2) );
+    return wheelRPM;
+  }
+
+  // get module yaw in RPM
+  public double motorRPMsToModuleYawRPM(double aMotorRPM, double bMotorRPM) {
+    // TODO: describe the math that got us here
+    double moduleYawRPM = ((aMotorRPM + bMotorRPM) / 2) / GEAR_RATIO_12;
+    return moduleYawRPM;
+  }
+
+  // convert desired translation and yaw RPMs to motor RPMs
+  public MotorRPMs getMotorSpeedsRPM(double wheelRPM, double yawRPM) {
+    double a = (yawRPM * GEAR_RATIO_12) + (wheelRPM * GEAR_RATIO_123);
+    double b = (yawRPM * GEAR_RATIO_12) - (wheelRPM * GEAR_RATIO_123);
+    return new MotorRPMs(a, b);
+  }
+
+  // create/initialize all of the shuffleboard inputs and outputs
+  private void initShuffleboard() {
+    sb_tab = Shuffleboard.getTab("Swerve");
+
+    sb_yawEnc       = sb_tab.getLayout("TurnEnc", BuiltInLayouts.kList);
+    sb_yawDist      = sb_yawEnc.add("distance", 0).getEntry();
+    sb_yawVelo      = sb_yawEnc.add("velocity", 0).getEntry();
+
+    sb_inputs       = sb_tab.getLayout("Inputs", BuiltInLayouts.kList);
+    sb_heading      = sb_inputs.add("heading",  0).getEntry();
+    sb_velocity     = sb_inputs.add("velocity", 0).getEntry();
+    sb_a_motorRPM   = sb_inputs.add("a motor RPM", 0).getEntry();
+    sb_b_motorRPM   = sb_inputs.add("b motor RPM", 0).getEntry();
+
+    sb_motorA       = sb_tab.getLayout("MotorA", BuiltInLayouts.kList);
+    sb_velA         = sb_motorA.add("velocity",  0).getEntry();
+    sb_velA_avg     = sb_motorA.add("velocity avg",  0).getEntry();
+
+    sb_motorB       = sb_tab.getLayout("MotorB", BuiltInLayouts.kList);
+    sb_velB         = sb_motorB.add("velocity",  0).getEntry();
+    sb_velB_avg     = sb_motorB.add("velocity avg",  0).getEntry();
+
+    sb_output       = sb_tab.getLayout("Output", BuiltInLayouts.kList);
+    sb_rot          = sb_output.add("Rotation",  0).getEntry();
+    sb_trans        = sb_output.add("Translation",  0).getEntry();
+
+    sb_debug        = sb_tab.getLayout("Debug", BuiltInLayouts.kList);
+    sb_use_motorRPM = sb_debug.add("Use Motor RPM", false).getEntry();
```

Comment:
unfortunately, network tables uses the string as the key so any changes here will require "fixing" the dashboard, not a big deal just a note

Reply:
I know, but I wanted to get some of these old things changed over